operator_ui/src/components/Bridges/Form.tsx

```diff
@@ -1,42 +1,63 @@
-import React from 'react'
-import PropTypes from 'prop-types'
-import { Prompt } from 'react-router-dom'
-import * as formik from 'formik'
-import { withStyles } from '@material-ui/core/styles'
-import { TextField, Grid } from '@material-ui/core'
+import { Grid, TextField } from '@material-ui/core'
+import {
+  createStyles,
+  Theme,
+  withStyles,
+  WithStyles
+} from '@material-ui/core/styles'
 import Button from 'components/Button'
-import { set, get } from 'utils/storage'
+import formik, { FormikProps } from 'formik'
 import normalizeUrl from 'normalize-url'
+import React from 'react'
+import { Prompt } from 'react-router-dom'
+import { get, set } from 'utils/storage'
 
-const styles = theme => ({
-  textfield: {
-    paddingTop: theme.spacing.unit * 1.25
-  },
-  card: {
-    paddingBottom: theme.spacing.unit * 2
-  },
-  button: {
-    marginTop: theme.spacing.unit * 3
-  },
-  flash: {
-    textAlign: 'center',
-    paddingTop: theme.spacing.unit,
-    paddingBottom: theme.spacing.unit
-  }
-})
+const styles = (theme: Theme) =>
+  createStyles({
+    textfield: {
+      paddingTop: theme.spacing.unit * 1.25
+    },
+    card: {
+      paddingBottom: theme.spacing.unit * 2
+    },
+    button: {
+      marginTop: theme.spacing.unit * 3
+    },
+    flash: {
+      textAlign: 'center',
+      paddingTop: theme.spacing.unit,
+      paddingBottom: theme.spacing.unit
+    }
+  })
 
-const isDirty = ({ values, submitCount }) => {
+const isDirty = ({ values, submitCount }: Props) => {
   return (
     (values.name !== '' ||
       values.url !== '' ||
-      (values.minimumContractPayment.toString() !== '0' &&
-        values.confirmations !== '0')) &&
+      (values.minimumContractPayment !== '0' && values.confirmations !== 0)) &&
     submitCount === 0
   )
 }
 
-const Form = props => (
-  <React.Fragment>
+interface OwnProps extends Partial<FormValues>, WithStyles<typeof styles> {
+  actionText: string
+  onSubmit: any
+  nameDisabled?: boolean
+  onSuccess: any
+  onError: any
+}
+
+interface FormValues {
+  name: string
+  minimumContractPayment: string
```

Comment:
Why is `minimumContractPayment` a `string` and `confirmations` a `number`? If anything, I think they should be the same type.

Reply:
These types are derived from the original `PropTypes` given. I havent changed anything as to not alter program logic. Wanted to keep all changes purely to the type system if possible first, so that its easier to review when we make possibly breaking changes.

Reply:
Agree with your reasoning. Worth noting that the original JS files will probably have lots of wrong PropTypes here and there, so now that we're `typescribing`, I feel like it might be better to make those changes now.
In this case confirmations and minContractPayment are both `input type="number"` so I think they should be typed as such too. 
Personally, I'd be making these changes now as I'd forget about them in future, but I'm good with whatever works for you the best.

Reply:
Ah I see, how about we make a compromise here and mark any converted `PropTypes` as `// CHECKME` so we can double back afterwards and fix any incorrect typings. That way we do not miss any, and we keep our PRs + git history easy to dissect since we'll separate type additions vs type fixes

Reply:
That's a good proposal, `// CHECKME` would make me happy with this kind of conversion commits. Will also make it easier for the reviewer to see through the intentions of the PR. Let's go with that if others don't chime in! :apple:

Reply:
sgtm!

Reply:
It looks like the server returns a string because it's a big int

![Screen Shot 2019-08-04 at 7 47 07 PM](https://user-images.githubusercontent.com/680789/62435359-b1a07500-b6f0-11e9-9583-da3a5bef48e5.png)